src/backend/api/meals.js

```diff
@@ -2,18 +2,129 @@ const express = require("express");
 const mealsRouter = express.Router();
 const dbcon = require("../database");
 
-// get all meals
+// get meals with query string
 mealsRouter.get("/", async (request, response) => {
-  let meals = undefined;
+  let mealQuery = dbcon("Meal");
+  // all the query strings
+  const { maxPrice } = request.query;
+  const { availableReservations } = request.query;
+  const { title } = request.query;
+  const { dateAfter } = request.query;
+  const { dateBefore } = request.query;
+  const { sortKey } = request.query;
+  const { sortDir } = request.query;
+  const { limit } = request.query;
+
+  // Condition check for maxPrice
+  if (maxPrice) {
+    if (maxPrice.match(/^\d+(.\d*)?$/)) {
+      mealQuery = mealQuery.where("price", "<=", maxPrice);
+    } else {
+      return response.status(400).json({
+        error: "No character only numeric value for maxPrice, please!",
+      });
+    }
+  }
+  // Condition check for availableReservations
+  if (availableReservations) {
+    if (availableReservations.toString().toLowerCase() === "true") {
+      mealQuery = mealQuery
+        .join("Reservation", {
+          "Meal.id": "Reservation.meal_id",
+        })
+        .select("Meal.id", "Meal.title", "Meal.max_reservations")
+        .sum({ already_reserved: "Reservation.number_of_guests" })
+        .groupBy("Reservation.meal_id")
+        .havingRaw("Meal.max_reservations > already_reserved");
```

Comment:
Try this `.having(knex.raw("Meal.max_reservations > already_reserved"))`.

Reply:
Thanks.